server/status.go

```diff
@@ -336,6 +341,82 @@ func (s *statusServer) handleLocalLog(w http.ResponseWriter, r *http.Request, ps
 	w.Write(b)
 }
 
+// handleLog checks the node_id parameter and if it matches that of the current
+// node, returns the local log entries. Otherwise, it looks up the specified
+// node in gossip and if it exists, sends a request directly to the node to get
+// its logs. It passes all other query parameters along to this new request.
+func (s *statusServer) handleLogs(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	nodeIDParam := ps.ByName("node_id")
+	if len(nodeIDParam) == 0 {
+		s.handleLocalLog(w, r, ps)
+		return
+	}
+
+	id, err := strconv.ParseInt(nodeIDParam, 10, 64)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "node id could not be parsed:%s", err)
+		return
+	}
+	nodeID := proto.NodeID(id)
+
+	if s.gossip.GetNodeID() == nodeID {
+		s.handleLocalLog(w, r, ps)
+		return
+	}
+
+	addr, err := s.gossip.GetNodeIDAddress(nodeID)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprint(w, "could not locate node")
+		return
+	}
+
+	// Create a call to the correct node.
+	tlsConfig, err := s.ctx.GetClientTLSConfig()
+	if err != nil {
+		log.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	httpClient := &http.Client{
+		Transport: &http.Transport{TLSClientConfig: tlsConfig},
+		Timeout:   logEntriesTimeout,
+	}
+
+	// Copy all the parameters to the new URL.
+	var params bytes.Buffer
+	for key, vals := range r.URL.Query() {
+		for _, val := range vals {
+			params.WriteString(fmt.Sprintf("%s=%s&", key, val))
+		}
+	}
+	url := fmt.Sprintf("%s://%s/_status/logs/?%s", s.ctx.RequestScheme(), addr, params.String())
+
+	// Call the other node.
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "error formatting internal request:%s", err)
+		return
+	}
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "there was an internal error:%s", err)
+		return
+	}
+	w.Header().Set(util.ContentTypeHeader, resp.Header.Get(util.ContentTypeHeader))
+	w.WriteHeader(resp.StatusCode)
```

Comment:
Why? We're only generating a few small responses right now. The most useful being a 400 when the parameters passed in are incorrect.